pymongo/topology_description.py

```diff
@@ -240,18 +240,21 @@ def apply_local_threshold(selection):
 
         if self.topology_type == TOPOLOGY_TYPE.Single:
             # Ignore the selector.
-            return self.known_servers
+            servers = self.known_servers
         elif address:
             description = self.server_descriptions().get(address)
-            return [description] if description else []
+            servers = [description] if description else []
         elif self.topology_type == TOPOLOGY_TYPE.Sharded:
             # Ignore the read preference, but apply localThresholdMS.
-            return apply_local_threshold(
+            servers = apply_local_threshold(
                 Selection.from_topology_description(self))
         else:
-            return apply_local_threshold(
+            servers = apply_local_threshold(
                 selector(Selection.from_topology_description(self)))
 
+        custom_selector = custom_selector or (lambda x: x)
```

Comment:
The spec requires that the custom selector is applied prior to applying the latency window check.

> 3: Find suitable servers by topology type and operation type
> 4: Filter the suitable servers by calling the optional, application-provided server selector.
> 5: If there are any suitable servers, choose one at random from those within the latency window and return it; otherwise, continue to the next step

Looks like here it's being applied after the latency window check and before the random selection of a single server.